web/init/src/components/shared/StepBuildingAssets.jsx

```diff
@@ -19,8 +24,10 @@ export default class StepBuildingAssets extends React.Component {
   }
 
   componentDidMount() {
-    const { startPollingStep, location, routeId } = this.props;
-    startPollingStep(location, routeId);
+    const { startPollingStep, currentRoute } = this.props;
+    if (currentRoute.phase === RENDER_PHASE) {
```

Comment:
This logic is here for these async steps to guard against re-polling. These components are re-rendered as the transition happens to the next phase as we're directly outputting JSX from the `renderStep` in `DetermineComponentForRoute`